Source/WebCore/rendering/RenderLayerModelObject.cpp

```diff
@@ -558,6 +586,34 @@ void RenderLayerModelObject::repaintOrRelayoutAfterSVGTransformChange()
     updateLayerTransform();
     repaintRendererOrClientsOfReferencedSVGResources();
 }
+
+void RenderLayerModelObject::paintSVGClippingMask(PaintInfo& paintInfo) const
+{
+    ASSERT(paintInfo.phase == PaintPhase::ClippingMask);
+    auto& context = paintInfo.context();
+    if (!paintInfo.shouldPaintWithinRoot(*this) || style().visibility() != Visibility::Visible || context.paintingDisabled())
+        return;
+
+    ASSERT(isSVGLayerAwareRenderer());
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    if (auto* referencedClipperRenderer = svgClipperResourceFromStyle())
+        referencedClipperRenderer->applyMaskClipping(paintInfo, *this, objectBoundingBox());
+#endif // ENABLE(LAYER_BASED_SVG_ENGINE)
+}
+
+void RenderLayerModelObject::paintSVGMask(PaintInfo& paintInfo, const LayoutPoint& adjustedPaintOffset) const
+{
+    auto& context = paintInfo.context();
+    if (!paintInfo.shouldPaintWithinRoot(*this) || paintInfo.phase != PaintPhase::Mask || context.paintingDisabled())
```

Comment:
IIRC we only call this for `PaintPhase::Mask`, no? Then you can add the same assertion as in `paintSVGClippingMask` just for `PaintPhase::Mask`.

Reply:
Added the ASSERT.